packages/node_modules/@webex/webex-core/src/interceptors/auth.js

```diff
@@ -90,23 +91,29 @@ export default class AuthInterceptor extends Interceptor {
       return Promise.resolve(true);
     }
 
-    // Validate that the service exists in the service catalog.
-    if (details || (service && services.hasService(service))) {
-      return Promise.resolve(true);
-    }
+    // Perform an additional validation in case the service does not exist yet.
+    return services.waitForService({name: service, url: uri})
+      .then((detectedUrl) => {
+        // Validate if the url was detected while waiting for the service.
+        if (detectedUrl) {
+          return true;
+        }
 
-    // Validate that the allowed domains can be utilized.
-    if (!services.validateDomains) {
-      Promise.resolve(false);
-    }
+        // Validate that the allowed domains can be utilized.
+        if (!services.validateDomains) {
+          return false;
+        }
 
-    // Validate that the domain of the uri is allowed.
-    if (services.hasAllowedDomains() &&
-      (uri && services.isAllowedDomainUrl(uri))) {
-      return Promise.resolve(true);
-    }
+        // Validate that the domain of the uri is allowed.
+        if (services.hasAllowedDomains() &&
+          (uri && services.isAllowedDomainUrl(uri))) {
+          return true;
+        }
 
-    return Promise.resolve(false);
+        // Return false to indicate authentication is not required.
+        return false;
+      })
+      .catch(() => false);
```

Comment:
I would remove this line. Let something else handle an exception or `Promise.reject`. If we insist on keeping it, we should at least log the error before returning `false`.

Reply:
I'm improving this catch statement to properly indicate via the logger the reason why the value was returned as false.